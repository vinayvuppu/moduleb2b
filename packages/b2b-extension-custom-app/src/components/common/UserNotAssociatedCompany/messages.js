import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'common.userNotAssociatedCompany.title',
    description: 'Title text for UserNotAssociatedCompany',
    defaultMessage: "Don't have any company",
  },
  paragraph1: {
    id: 'common.userNotAssociatedCompany.paragraph1',
    description: 'Paragraph text for UserNotAssociatedCompany',
    defaultMessage:
      'Your user with email {email} is not associated to any company',
  },
});
