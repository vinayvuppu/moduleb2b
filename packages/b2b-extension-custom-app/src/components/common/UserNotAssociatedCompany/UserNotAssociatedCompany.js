import React from 'react';
import { MaintenancePageLayout } from '@commercetools-frontend/application-components';
import UserNotAssociatedSVG from '@commercetools-frontend/assets/images/doors-closed.svg';
import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
import { useIntl } from 'react-intl';
import messages from './messages';

const UserNotAssociatedCompany = () => {
  const { formatMessage } = useIntl();
  const {
    user: { email },
  } = useApplicationContext();

  return (
    <MaintenancePageLayout
      imageSrc={UserNotAssociatedSVG}
      title={formatMessage(messages.title)}
      paragraph1={formatMessage(messages.paragraph1, { email })}
    ></MaintenancePageLayout>
  );
};

UserNotAssociatedCompany.displayName = 'UserNotAssociatedCompany';

export default UserNotAssociatedCompany;
