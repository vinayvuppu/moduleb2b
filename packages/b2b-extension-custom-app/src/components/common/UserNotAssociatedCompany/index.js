import UserNotAssociatedCompany from './UserNotAssociatedCompany';

export default UserNotAssociatedCompany;
