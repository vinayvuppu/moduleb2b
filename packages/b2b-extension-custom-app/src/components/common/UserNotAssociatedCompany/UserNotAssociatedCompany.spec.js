import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import UserNotAssociatedCompany from './UserNotAssociatedCompany';

jest.mock('@commercetools-frontend/application-shell-connectors', () => ({
  useApplicationContext: () => ({ user: { email: 'foo@example.com' } }),
}));

afterEach(cleanup);

describe('UserNotAssociatedCompany component', () => {
  describe('rendering', () => {
    it('should render the correct snapshot', () => {
      const { asFragment } = render(
        <IntlProvider locale="en">
          <UserNotAssociatedCompany />
        </IntlProvider>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
