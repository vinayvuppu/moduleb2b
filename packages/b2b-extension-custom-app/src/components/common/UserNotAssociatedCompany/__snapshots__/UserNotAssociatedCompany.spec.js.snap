// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNotAssociatedCompany component rendering should render the correct snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-3mrj0b"
  >
    <div
      class="css-i7yiek-Horizontal-getConstraintStyles-Horizontal"
    >
      <div
        class="css-y97dwm-getStyles"
      >
        <div>
          <img
            src="test-file-stub"
          />
        </div>
        <h2
          class="css-1d7aioo-headlineStyles"
        >
          Don't have any company
        </h2>
        <p
          class="css-f4iurr-bodyStyles"
        >
          Your user with email foo@example.com is not associated to any company
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
